base/collect_http.go

```diff
@@ -218,14 +244,67 @@ func getFortioOptions(c endpoint) (*fhttp.HTTPRunnerOptions, error) {
 	return fo, nil
 }
 
+func putPerformanceResultToMetricsService(metricsServerURL, namespace, experiment string, data interface{}) error {
+	// handle URL and URL parameters
+	u, err := url.ParseRequestURI(metricsServerURL + PerformanceResultPath)
+	if err != nil {
+		return err
+	}
+
+	params := url.Values{}
+	params.Add("namespace", namespace)
+	params.Add("experiment", experiment)
+	u.RawQuery = params.Encode()
+	urlStr := fmt.Sprintf("%v", u)
+
+	log.Logger.Trace(fmt.Sprintf("PUT request URL: %s", urlStr))
+
+	// handle payload
+	dataBytes, err := json.Marshal(data)
+	if err != nil {
+		log.Logger.Error("cannot JSON marshal data for metrics server request: ", err)
+		return err
+	}
+
+	// create request
+	req, err := http.NewRequest(http.MethodPut, urlStr, bytes.NewBuffer(dataBytes))
+	if err != nil {
+		log.Logger.Error("cannot create new HTTP request metrics server: ", err)
+		return err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	log.Logger.Trace("sending request")
+
+	// send request
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Logger.Error("could not send request to metrics server: ", err)
+		return err
+	}
+	defer func() {
+		err = resp.Body.Close()
+		if err != nil {
+			log.Logger.Error("could not close response body: ", err)
+		}
+	}()
+
+	log.Logger.Trace("sent request")
+
+	return nil
+}
+
 // getFortioResults collects Fortio run results
 // func (t *collectHTTPTask) getFortioResults() (*fhttp.HTTPRunnerResults, error) {
 // key is the metric prefix
-func (t *collectHTTPTask) getFortioResults() (map[string]*fhttp.HTTPRunnerResults, error) {
+// key is the endpoint
+func (t *collectHTTPTask) getFortioResults() (map[string]fhttp.HTTPRunnerResults, error) {
```

Comment:
since we are attempting to keep the old functionality, I think it better to leave the data structure the same.